provisioning/baremetal_config.go

```diff
@@ -0,0 +1,116 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"fmt"
+
+	metal3iov1alpha1 "github.com/openshift/cluster-baremetal-operator/api/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	// BaremetalProvisioningCR is the name of the provisioning resource
+	BaremetalProvisioningCR = "provisioning-configuration"
```

Comment:
Requiring this specific name is a characteristic of the controller, rather than the resource type, right? It seems like the only use inside this package are for the logger, and we could skip that, and the unit tests where it shouldn't matter. I think I would leave it back in the controller package.

Reply:
It has been moving around. Yes, it makes sense that the name of the resource is a function of the controller. Made the necessary changes.